### ISSUE TITLE
Wrong error TAG-336-0002 "DotRange invalid cardinality"

### ISSUE THREAD

**Issue by marhop** [EasyinnovaSL/DPFManager#479]
Hi,

when validating some files using the Extended TIFF 6.0 profile, DPF Manager 3.5.1 throws this error:

> TAG-336-0002 "DotRange invalid cardinality, DotRange Tag cardinality must be equal 2 or 2 * BitsPerSample"

This is wrong. The condition should say "DotRange Tag cardinality must be equal 2 or 2 * *SamplesPerPixel*" (not BitsPerSample). I guess two conditions from the The TIFF 6.0 specification got messed up here. On page 71 it requires that:

 1. The DotRange tag holds an array with either 2 or (2 * SamplesPerPixel) entries.
 2. Each entry in this array has a value within the range from 0 to (2 ^ BitsPerSample - 1).

---

However, I think it is not just a wrong error message but the conformance checks themselves seem to be messed up a little bit. For example, the offending tag in one of my files looks like this in hex (little endian):

    5001 0100 02000000 00ff0000

In the same file SamplesPerPixel has value 4 and BitsPerSample has value 8. This tag says:

  * Its tag number is 0x0150 (336 DotRange).
  * Its items have type 0x01 (unsigned byte).
  * It holds 0x02 (2) items in the tag data.
  * The first item has value 0x00 (0) and the second item has value 0xff (255).

So the tag holds an array of length 2 (which satisfies the first requirement) and both its items have values within the range from 0 to 255 = 2 ^ 8 - 1 = 2 ^ BitsPerSample - 1 (which satisfies the second requirement). Why does DPF Manager throw an error at all (regardless of error message)?

**Comment by victormunoz** [comment]
Yes you are right. Both the message and the condition that is checked is wrong. It should look for the SamplesPerPixel tag instead of the BitsPerSample for the DotRange cardinality check.
This bug is going to be fixed in the next release.
Thank you very much for reporting!

**Comment by marhop** [comment]
That's good to hear, thank you! I will test again with the new release.
